### ISSUE TITLE
HamburgerMenu background doesn't update on theme change

### ISSUE THREAD

**Issue by janabimustafa** [Windows-XAML/Template10#355]
I have the following style code

``` XAML
<Style TargetType="controls:HamburgerMenu">
        <Setter Property="NavButtonCheckedBackground" Value="{StaticResource BlueBrush}"/>
        <Setter Property="NavAreaBackground" Value="{ThemeResource ChromeBrush}"/>        
        <Setter Property="VisualStateNarrowMinWidth" Value="{StaticResource NarrowMinWidth}" />
        <Setter Property="VisualStateNormalMinWidth" Value="{StaticResource NormalMinWidth}" />
        <Setter Property="VisualStateWideMinWidth" Value="{StaticResource WideMinWidth}" />
    </Style>
    <Style TargetType="controls:PageHeader">
        <Setter Property="Background" Value="{ThemeResource ChromeBrush}"/>
        <Setter Property="VisualStateNarrowMinWidth" Value="{StaticResource NarrowMinWidth}" />
        <Setter Property="VisualStateNormalMinWidth" Value="{StaticResource NormalMinWidth}" />
    </Style>
    <ResourceDictionary.ThemeDictionaries>
        <ResourceDictionary x:Key="Dark">
            <SolidColorBrush x:Key="ChromeBrush" Color="Red"/>
        </ResourceDictionary>
        <ResourceDictionary x:Key="Light">
            <SolidColorBrush x:Key="ChromeBrush" Color="Green"/>
        </ResourceDictionary>
    </ResourceDictionary.ThemeDictionaries>
```

This is the result. 
![HamburgerMenu background](http://i.imgur.com/N8nvChA.gif)

I don't want to move the declaration of the style inside the ThemeDictionaries (like it is in the Minimal template) as that goes against the [ThemeDictionaries Guidelines](https://msdn.microsoft.com/en-us/library/windows/apps/windows.ui.xaml.resourcedictionary.themedictionaries.aspx) and it doesn't fix the issue as a restart of the app will be required for the theme to apply.

**Comment by JerryNixon** [comment]
Quote the rule we're breaking. I'll ask the team.

**Comment by janabimustafa** [comment]
> When the app first starts, any resource reference made by a ThemeResource reference is evaluated based on the theme in use at startup. **But if the user subsequently changes the active theme at run-time, the system will re-evaluate every ThemeResource reference**, retrieve a theme-specific resource that may be different, and redisplay the app with new resource values in all appropriate places in the visual tree. 
> [From this page](https://msdn.microsoft.com/en-us/library/windows/apps/mt185591.aspx)

In code:
This will work (I've tested it):

``` XAML
<ResourceDictionary.ThemeDictionaries>
        <!--  RequestedTheme=Dark  -->
        <ResourceDictionary x:Key="Default">
            <Style x:Key="PageHeaderStyle" TargetType="controls:PageHeader">
                <Setter Property="Background" Value="Red" />
            </Style>
        </ResourceDictionary>
        <!--  RequestedTheme=Light  -->
        <ResourceDictionary x:Key="Light">
            <Style x:Key="PageHeaderStyle" TargetType="controls:PageHeader">
                <Setter Property="Background" Value="Green" />
            </Style>
        </ResourceDictionary>
And then using it like this
        <controls:PageHeader Style="{ThemeResource PageHeaderStyle}"  Content="Settings Page" Frame="{x:Bind Frame}" />
```

And this won't work (current implementation)

``` XAML
<ResourceDictionary.ThemeDictionaries>
        <!--  RequestedTheme=Dark  -->
        <ResourceDictionary x:Key="Default">
            <Style TargetType="controls:PageHeader">
                <Setter Property="Background" Value="Red" />
            </Style>
        </ResourceDictionary>
        <!--  RequestedTheme=Light  -->
        <ResourceDictionary x:Key="Light">
            <Style TargetType="controls:PageHeader">
                <Setter Property="Background" Value="Green" />
            </Style>
        </ResourceDictionary>
And then using it like this
        <controls:PageHeader Content="Settings Page" Frame="{x:Bind Frame}" />
```

Since a **ThemeResource** is never used in the second example, it will not be reevaluated when the theme changes.

**Comment by dg2k** [comment]
The theme updating can be done entirely within the Minimal project template. I did this in my PR #195 which I closed in favor of your PR #202 (PageHeader Templated Control). Looks like we're going in circles ;-)

The code that does the trick by accessing the custom.xaml content is in shell.xaml.cs which I believe you can still access in PR #195. Shown below is a demo.

![hammenuthemerefresh](https://cloud.githubusercontent.com/assets/14026786/11318811/f853615a-9058-11e5-82d7-d6ed02fad644.gif)

**Comment by JerryNixon** [comment]
Now that you have shown me what content you are quoting, I can confidently tell you that this is not a rule and that we are not breaking it. The docs are describing the behavior of theme resources. We are using them as they are designed to be used. You should use them as we are using them because this is the text-book approach and is correct.
